parse/NamedValueRefParser.cpp

```diff
@@ -0,0 +1,131 @@
+                const std::string& filename,
+                const parse::text_iterator& first,
+                const parse::text_iterator& last) :
+            grammar::base_type(start, "named_value_ref_grammar"),
+            condition_parser(tok, label),
+            string_grammar(tok, label, condition_parser),
+            int_rules(tok, label, condition_parser, string_grammar),
+            double_rules(tok, label, condition_parser, string_grammar)
+        {
+            namespace phoenix = boost::phoenix;
+            namespace qi = boost::spirit::qi;
+
+            qi::_1_type _1;
+            qi::_2_type _2;
+            qi::_3_type _3;
+            qi::_4_type _4;
+            qi::_val_type _val;
+            qi::_pass_type _pass;
+            qi::omit_type omit_;
+            qi::_r1_type _r1;
+
+            named_ref
+               =
+                    ( omit_[tok.Named_]   >> tok.Integer_
+                   > label(tok.Name_) > tok.string
+                   > label(tok.Value_) >  qi::as<parse::detail::MovableEnvelope<ValueRef::ValueRef<int>>>()[int_rules.expr]
+                    ) [ insert_named_ref_(_r1, _2, _3, _pass) ]
+                    |
+                    ( omit_[tok.Named_]   >> tok.Real_
+                   > label(tok.Name_) > tok.string
+                   > label(tok.Value_) >  qi::as<parse::detail::MovableEnvelope<ValueRef::ValueRef<double>>>()[double_rules.expr]
+                    ) [ insert_named_ref_(_r1, _2, _3, _pass) ]
+                ;
+
+
+            start
+              = +named_ref(_r1)
+                ;
+
+            named_ref.name("NamedValueRef");
+
+#if DEBUG_PARSERS
+            debug(named_ref);
+#endif
+
+	    qi::on_error<qi::fail>(start, parse::report_error(filename, first, last, _1, _2, _3, _4));
+        }
+
+        using named_value_ref_rule = parse::detail::rule<
+            void (std::map<std::string, std::unique_ptr<ValueRef::AnyValueRef>>&)>;
+
+        using start_rule = parse::detail::rule<start_rule_signature>;
+
+        parse::detail::Labeller                 label;
+        const parse::conditions_parser_grammar  condition_parser;
+        const parse::string_parser_grammar      string_grammar;
+        const parse::int_arithmetic_rules       int_rules;
+        const parse::double_parser_rules        double_rules;
+        named_value_ref_rule                    named_ref;
+        start_rule                              start;
+    };
+}
+
+namespace parse {
+    start_rule_payload named_value_refs(const boost::filesystem::path& path) {
+        const lexer lexer;
+        start_rule_payload named_value_refs;
+
+        ScopedTimer timer("Named ValueRef Parsing", true);
+
+        for (const auto& file : ListDir(path, IsFOCScript))
+            detail::parse_file<grammar, start_rule_payload>(lexer, file, named_value_refs);
+
+        for (auto& k_v : named_value_refs )
+                ErrorLogger() << "named_value_refs : " << k_v.first;
```

Comment:
4 spaces indent

Reply:
i will do an extra pass at the end through all the added logging. currently i need the visibility of such output in the log files. so, later probably DebugLogger here